langchain/output_parsers/json.py

```diff
@@ -27,15 +27,32 @@ def parse_json_markdown(json_string: str) -> dict:
     return parsed
 
 
-def parse_and_check_json_markdown(text: str, expected_keys: List[str]) -> dict:
+def parse_and_check_json_markdown(
+    text: str,
+    expected_keys: List[str]
+) -> Union[dict, List[dict]]:
     try:
         json_obj = parse_json_markdown(text)
     except json.JSONDecodeError as e:
         raise OutputParserException(f"Got invalid JSON object. Error: {e}")
-    for key in expected_keys:
-        if key not in json_obj:
-            raise OutputParserException(
-                f"Got invalid return object. Expected key `{key}` "
-                f"to be present, but got {json_obj}"
-            )
+    if isinstance(json_obj, list):
+      expected_keys_set = set(expected_keys)
+      missing_keys = []
+
+      for record in json_obj:
+            missing = expected_keys_set - set(record.keys())
+            missing_keys.extend(missing)
+      
+      if missing_keys:
+        raise OutputParserException(
+            f"Got invalid return object. Expected key(s) `{missing_keys}` "
+            f"to be present, but got {record}"
```

Comment:
This could be unset if the list is empty

Reply:
If the `expected_keys` is empty? Not sure what you mean